src/commands/stack/add.ts

```diff
@@ -0,0 +1,185 @@
+    }
+
+    let databaseName: string =
+      flags.database ??
+      (await this.promptDatabasePath(
+        config.rootConfig.endpoints[endpointName]
+      ));
+
+    const validateStackName = (input: string) => {
+      if (input.length === 0) {
+        return "Stack name cannot be empty";
+      } else if (
+        Object.keys(config.projectConfig?.stacks ?? {}).includes(input)
+      ) {
+        return `Stack ${input} already exists`;
+      } else {
+        return true;
+      }
+    };
+
+    const name =
+      flags.name ??
+      (await input({
+        message: "Stack name",
+        validate: validateStackName,
+      }));
+    const res = validateStackName(name);
+    if (res !== true) {
+      this.error(res);
+    }
+
+    const setDefault =
+      flags["set-default"] ??
+      (flags["non-interactive"]
+        ? false
+        : await confirm({
+            message: "Make this stack default",
+          }));
+
+    if (setDefault) {
+      config.projectConfig.defaultStack = name;
+    }
+
+    config.projectConfig.stacks[name] = {
+      endpoint: endpointName,
+      database: databaseName,
+    };
+    config.saveProjectConfig();
+    console.log(`Saved stack ${name} to ${config.projectConfigFile()}`);
+  }
+
+  promptDatabasePath = async (endpoint: Endpoint): Promise<string> => {
+    const { url, secret } = endpoint;
+    const client = new FaunaClient({ endpoint: url, secret });
+
+    const res = await client.query("0");
+    if (res.status != 200) {
+      this.error(`Error: ${res.body.error.code}`);
+    }
+
+    const databasePaths = await this.getDatabasePaths(client);
+
+    client.close();
+
+    return await searchSelect({
+      message: "Select a database",
+      choices: databasePaths.map((database) => ({
+        value: database,
+      })),
+    });
+  };
+
+  getDatabasePaths = async (client: FaunaClient): Promise<string[]> => {
+    // Limits: choose a limit of 100 databases at each depth, and a depth of 10.
+    // We will also add a limit if any databases are skiped.
+    const database_limit = 100;
+    const depth_limit = 10;
+
+    const allDatabases: string[] = [];
+    const paths: string[][] = [[]];
+
+    let overflowedLimit = false;
+    let overflowedDepth = false;
+
+    ux.action.start("Fetching databases");
+
+    while (true) {
+      const path = paths.pop();
+      if (path === undefined) {
+        break;
+      }
+      if (path.length > depth_limit) {
+        overflowedDepth = true;
+        continue;
+      }
+
+      const databases = await client.query<string[]>(
+        `Database.all().take(${database_limit + 1}).toArray().map(.name)`,
+        {
+          secret: `${client.secret}:${path.join("/")}:admin`,
+        }
+      );
+      if (databases.status !== 200) {
+        this.error(`Error: ${databases.body.error.code}`);
+      }
+
+      if (databases.body.data.length > database_limit) {
+        overflowedLimit = true;
```

Comment:
idt we currently use this anywhere, but as discussed offline, we want to rework the limit situation a bit, I went ahead and just created this jira:
https://faunadb.atlassian.net/browse/ENG-5574
to track that

Reply:
perfect. this bool will be used later to determine if we should just let the user write in a path.